query-engine/connectors/sql-query-connector/src/error.rs

```diff
@@ -59,6 +61,13 @@ impl From<quaint::error::Error> for RawError {
                 code: e.original_code().map(ToString::to_string),
                 message: e.original_message().map(ToString::to_string),
             },
+        };
+
+        // note: this additional match is needed to move Box<dyn std::error::Error + Send + Sync + 'static> into any RawError
+        // that needs to wrap the Box, e.g., RawError::QueryInvalidInput
+        match QuaintKind::from(e) {
+            quaint::error::ErrorKind::QueryInvalidInput(qe) => Self::QueryInvalidInput(qe),
+            _ => raw_error,
         }
```

Comment:
Please let me know whether this is an acceptable solution for the problem highlighted in the note.